src/hotspot/cpu/riscv/sharedRuntime_riscv.cpp

```diff
@@ -1473,8 +1471,8 @@ nmethod* SharedRuntime::generate_native_wrapper(MacroAssembler* masm,
 
   if (VM_Version::supports_fast_class_init_checks() && method->needs_clinit_barrier()) {
     Label L_skip_barrier;
-    __ mov_metadata(t1, method->method_holder()); // InstanceKlass*
-    __ clinit_barrier(t1, t0, &L_skip_barrier);
+    __ mov_metadata(t0, method->method_holder()); // InstanceKlass*
+    __ clinit_barrier(t0, t1, &L_skip_barrier);
```

Comment:
See there is still one odd one here, sharedRuntime_riscv.cpp L2849.
I can revert or change that one also ?

Reply:
Ah, I see. I would prefer to keep the original shape. As you see, the first param for MacroAssembler::clinit_barrier is `klass` which is supposed to be alive for quite a range. I think `t1` is safer here because `t0` is implictly clobbered by various assember routines.